scripts/createTsIconComponents.js

```diff
@@ -0,0 +1,240 @@
+
+const rootDir = path.join(__dirname, '..', 'packages', 'ndla-icons');
+const attrs = ['xlink:href', 'id'];
+
+const copyright = fs.readFileSync(
+  path.join(__dirname, '..', 'COPYRIGHT'),
+  'utf-8',
+);
+const autoNotice = '// N.B! AUTOGENERATED FILE. DO NOT EDIT';
+
+function capitalize(string) {
+  return string.charAt(0).toUpperCase() + string.substring(1);
+}
+
+function ensureDirectoryExistence(filePath) {
+  const dirname = path.dirname(filePath);
+  if (fs.existsSync(dirname)) {
+    return;
+  }
+  ensureDirectoryExistence(dirname);
+  fs.mkdirSync(dirname);
+}
+
+function cleanAttributes($el, $) {
+  attrs.forEach(attr => {
+    $el.removeAttr(attr);
+  });
+  if ($el.children().length === 0) {
+    return false;
+  }
+
+  $el.children().each((index, el) => {
+    cleanAttributes($(el), $);
+  });
+  return true;
+}
+
+function camelCaseAttributes($el, $) {
+  Object.keys($el[0].attribs).forEach(attr => {
+    if (
+      $el[0].name !== 'svg' &&
+      attr.indexOf('-') > 0 &&
+      !attr.startsWith('data-')
+    ) {
+      const value = $el[0].attribs[attr];
+      $el.removeAttr(attr);
+      const camelCaseAttr = attr
+        .split('-')
+        .map((token, i) => (i === 0 ? token : capitalize(token)))
+        .join('');
+      $el.attr(camelCaseAttr, value);
+    }
+  });
+
+  if ($el.children().length === 0) {
+    return false;
+  }
+
+  $el.children().each((index, el) => {
+    camelCaseAttributes($(el), $);
+  });
+  return true;
+}
+
+function createComponent(name, svg) {
+  const $ = cheerio.load(svg, {
+    xmlMode: true,
+  });
+  const $svg = $('svg');
+  const viewBox = $svg.attr('viewBox');
+  const license = $svg.attr('data-license');
+  const source = $svg.attr('data-source');
+  if (!license || !source) {
+    return undefined;
+  }
+
+  cleanAttributes($svg, $);
+  camelCaseAttributes($svg, $);
+
+  const iconSvg = $svg.html();
+  return prettier.format(
+    `${copyright}
+${autoNotice}
+import React, { ReactNode } from 'react';
+import Icon from '../Icon';
+
+interface Props {
+  children?: ReactNode;
+}
+
+const ${name} = (props: Props) => (
+  <Icon
+    title="${name}"
+    viewBox="${viewBox}"
+    data-license="${license}"
+    data-source="${source}"
+    {...props}>
+    <g>
+      ${iconSvg}
+    </g>
+  </Icon>
+);
+
+export default ${name};
+`,
+    {
+      ...prettierOptions,
+      parser: 'babylon',
+    },
+  );
+}
+
+function writeIndexFiles(types) {
+  Object.keys(types).forEach(folder => {
+    const components = types[folder];
+    const exportsString = components
+      .map(
+        component =>
+          `export { default as ${component} } from './${component}';`,
+      )
+      .join('\n');
+    const iconsModule = `${copyright}
+${autoNotice}
+${exportsString}\n`;
+    const formatedIconsModule = prettier.format(iconsModule, {
+      ...prettierOptions,
+      parser: 'babylon',
+    });
+    const fileName = path.join(rootDir, 'src', folder, 'index.js');
```

Comment:
Skulle ikjke denne også blitt ts?
```suggestion
    const fileName = path.join(rootDir, 'src', folder, 'index.ts');
```